src/test/kotlin/org/rust/ide/actions/RsJoinRawLinesHandlerTest.kt

```diff
@@ -4,12 +4,90 @@ class RsJoinRawLinesHandlerTest : RsJoinLinesHandlerTestBase() {
     fun `test block expression`() = doTest("""
         fn main() {
             let _ = /*caret*/{
-                92
+                42
             };
         }
     """, """
         fn main() {
-            let _ = /*caret*/92;
+            let _ = /*caret*/42;
+        }
+    """)
+
+    fun `test join struct selection`() = doTest("""
+        struct S { foo: i32, bar: i32 }
+        fn main() {
+            let _ = S <selection>{
+                foo: 42,
+                bar: 42,
+            };</selection>
+        }
+    ""","""
+        struct S { foo: i32, bar: i32 }
+        fn main() {
+            let _ = S { foo: 42, bar: 42 };
+        }
+    """)
+
+    fun `test join struct`() = doTest("""
+        struct S { foo: i32 }
+        fn main() {
+            let _ = S { /*caret*/
+                foo: 42,
+            };
+        }
+    ""","""
+        struct S { foo: i32 }
+        fn main() {
+            let _ = S { foo: 42,
+            };
+        }
+    """)
+
+    fun `test remove comma 1`() = doTest("""
+        struct S { foo: i32 }
+        fn main() {
+            let _ = S { foo: 42, /*caret*/
+             };
+        }
+    ""","""
+        struct S { foo: i32 }
+        fn main() {
+            let _ = S { foo: 42 };
+        }
+    """)
+
+    fun `test remove comma 2`() = doTest("""
+        struct S { foo: i32, bar: i32 }
+        fn main() {
+            let _ = S {
+                foo: 42,
+                bar: /*caret*/42,
+             };
+        }
+    ""","""
+        struct S { foo: i32, bar: i32 }
+        fn main() {
+            let _ = S {
+                foo: 42,
+                bar: 42/*caret*/
+            };
```

Comment:
This seems a bit strange: we've removed the comma, but we've failed to actually join the lines...